### PR TITLE
[TKW] Polish up reduction tests

### PR DESCRIPTION
**Pull request by raikonenfnu**

### CODE REVIEW

File: tests/kernel/wave/wave_e2e_test.py [hunk 1]
```diff
@@ -290,7 +291,7 @@ def test(
         run_config=config,
     ):
         test(a, b, c)
-        assert_allclose(ref.values, c)
+        assert torch.equal(c, ref.values)
```
Comment (Hardcode84) [inline tests/kernel/wave/wave_e2e_test.py:294]: Do we expect tensors to be bit-exact? Why not assert_equal?

Comment (raikonenfnu) [reply tests/kernel/wave/wave_e2e_test.py:294]: for max, yes :)

Comment (Hardcode84) [reply tests/kernel/wave/wave_e2e_test.py:294]: Oh, wait, I didn't noticed it was max ), anyway, why not `assert_equal`?

Comment (raikonenfnu) [reply tests/kernel/wave/wave_e2e_test.py:294]: Keeping it uniform with the wave_e2e_gemm test

Comment (Hardcode84) [reply tests/kernel/wave/wave_e2e_test.py:294]: `assert_equal` is better, it will actually print array contents and number of elements mismatched.

Comment (raikonenfnu) [reply tests/kernel/wave/wave_e2e_test.py]: Done!

**Review by Hardcode84 — APPROVED** [review]

**Pull request merged**